### PR TITLE
cfgplugins optical changes + Cisco deviations

### PR DESCRIPTION
**Pull request by karthikeya-remilla**
this PR builds on top of code changes proposed by @hattikals over multiple open PRs.

Changes and brief description for the following files:

1. zr_cd_test: use cfgplugins code + enable operational-mode flag
2. zr_input_output_test: enable operational-mode flag
3. cfgplugins/interface.go: update the common function InterfaceConfig for tests to consume

### CODE REVIEW

**Review by ahsaanyousaf — APPROVED** [review]

File: feature/platform/transceiver/tests/zr_cd_test/zr_cd_test.go
Comment (ejbrever) [inline feature/platform/transceiver/tests/zr_cd_test/zr_cd_test.go:35]: just to make it explicit, can you please call this "operationalModeFlag" in all files?

File: internal/cfgplugins/interface.go
Comment (ejbrever) [inline internal/cfgplugins/interface.go:38]: Oh interesting....we actually just went through this in a different place for this PR:
https://github.com/openconfig/featureprofiles/pull/2891

https://github.com/hattikals/featureprofiles/blob/3a34ffd7466037956b25ddea4ffbe34aae09e7ee/internal/components/components.go#L203

I don't know why we have this same code in two places :/

We should probably merge this somehow and only have one. Maybe just to start in this PR, can you just copy the logic from the other file?

File: feature/platform/transceiver/tests/zr_temperature_test/zr_temperature_test.go
Comment (ejbrever) [inline feature/platform/transceiver/tests/zr_temperature_test/zr_temperature_test.go:96]: can we rewrite this a little for readability:

var compWithTemperature *oc.Component
switch dut1.Vendor() {
case ondatra.CISCO:
  compWithTemperature = gnmi.OC().Component(transceiverName)
default:
  <...add the for loop logic here...>
}

**Review by ejbrever — COMMENTED** [review]
I see your note about once getting conditional approval you plan to change more files. Please do not do this.

It really helps to keep changes as small as possible for quick reviews. In the future if you can break the PRs up into small pieces that will be very helpful.

File: feature/platform/transceiver/tests/zr_temperature_test/zr_temperature_test.go
Comment (dplore) [reply feature/platform/transceiver/tests/zr_temperature_test/zr_temperature_test.go:96]: Also note, the canonical OC to get temperature is through a subcomponent sensor.  If cisco does not support this, then a deviation should be defined and used instead of `if dut1.Vendor() != ondatra.CISCO `

File: internal/cfgplugins/interface.go
Comment (dplore) [reply internal/cfgplugins/interface.go:38]: I recommended this change offline.  Yes, the goal should be to migrate to using `internal/cfgplugins` for common configuration functions.  For ease of review, all the migration does not need to happen in this PR.

Comment (ejbrever) [reply internal/cfgplugins/interface.go:38]: Agree, we should definitely break that up in multiple PRs.

**Comment by karthikeya-remilla** [comment]
> I see your note about once getting conditional approval you plan to change more files. Please do not do this.
> 
> It really helps to keep changes as small as possible for quick reviews. In the future if you can break the PRs up into small pieces that will be very helpful.

@ejbrever @dplore I agree about keeping the PR small for ease of review. I assumed that all files must be migrated in this PR to not break regression runs, since we are changing the common utility function. 

Will split this PR by reducing the changed files. Thanks!

File: feature/platform/transceiver/tests/zr_temperature_test/zr_temperature_test.go
Comment (karthikeya-remilla) [reply feature/platform/transceiver/tests/zr_temperature_test/zr_temperature_test.go:96]: Ack - I'll push these changes for zr_temperature_test in my next PR.

File: internal/cfgplugins/interface.go
Comment (karthikeya-remilla) [reply internal/cfgplugins/interface.go:38]: Ack - will revert to the tests using components.OpticalChannelComponentFromPort approved as part of #2891

**Review by ejbrever — APPROVED** [review]
Thanks for the changes!

**Review by ahsaanyousaf — APPROVED** [review]

**Pull request merged**
